cmd/clusterctl/client/cluster/inventory.go

```diff
@@ -103,6 +103,10 @@ type InventoryClient interface {
 	// does not match the current one supported by clusterctl.
 	CheckCAPIContract(...CheckCAPIContractOption) error
 
+	// CheckCAPIInstalled checks if Cluster API is installed on the management cluster. Returns an error if an incompatible
+	// version is installed.
+	CheckCAPIInstalled() (bool, error)
```

Comment:
This does not seem to be used in this PR so I suggest to remove it from this PR.

Also, it seems conflating two problems: a) CAPI is installed b) CAPI version supports the expected ClusterAPI contract version

If we need b, we can use `CheckCAPIContract` directly without a wrapper. if we need a only, we can't use `CheckCAPIContract` as it is. If you want to re-use it, we could consider to make it possible something like:

```
if CheckCAPIContract(
		cluster.AllowCAPIContract{Contract: "*"},
	) ...
```

Reply:
> This does not seem to be used in this PR so I suggest to remove it from this PR.

Is is okay to keep the implementation in the PR and just drop the function from the interface?


> Also, it seems conflating two problems: a) CAPI is installed b) CAPI version supports the expected ClusterAPI contract version.

We are looking for option A. If we choose to re-use CheckCAPIContract internally then are there any preferences between having `cluster.AllowCAPIContract{Contract: "*"}`  and `cluster.AllowCAPIAnyContract{}` (similar to `cluster.AllowCAPINotInstalled{}`)?

Reply:
`cluster.AllowCAPIAnyContract{}` makes sense to me

Reply:
> Is is okay to keep the implementation in the PR and just drop the function from the interface?

Let's make this PR consistent, containing only what is required to solve the problem at stake